account_payment_term_discount/views/account_payment_term_view.xml

```diff
@@ -0,0 +1,38 @@
+<odoo>
+    <!-- Inherit Tree View Account Payment Term -->
+    <record id="view_payment_term_disc_tree" model="ir.ui.view">
+        <field name="name">account.payment.term.disc.tree</field>
+        <field name="model">account.payment.term</field>
+        <field name="inherit_id" ref="account.view_payment_term_tree" />
+        <field name="arch" type="xml">
+            <field name="name" position="after">
+                <field name="is_discount" />
+                <field name="is_exclude_shipping_lines" />
+                <field name="is_exclude_taxes_discount" />
+                <field name="active" />
+            </field>
+        </field>
+    </record>
+
+    <!-- Inherit Form View Account Payment Term -->
+    <record id="view_payment_term_disc_form" model="ir.ui.view">
+        <field name="name">account.payment.term.disc.form</field>
+        <field name="model">account.payment.term</field>
+        <field name="inherit_id" ref="account.view_payment_term_form" />
+        <field name="arch" type="xml">
+            <xpath expr="/form/sheet/group/group/field[@name='name']" position="after">
+                <field name="is_discount" />
+                <field name="is_exclude_shipping_lines" />
+                <field name="is_exclude_taxes_discount" />
+            </xpath>
```

Comment:
```suggestion
            <xpath expr="/form/sheet/group/group/field[@name='name']" position="after">
                <field name="is_discount" />
                <field name="is_exclude_shipping_lines" attrs="{'invisible': [('is_discount', '=', False)]}"/>
                <field name="is_exclude_taxes_discount" attrs="{'invisible': [('is_discount', '=', False)]}" />
            </xpath>
```
For UX purposes, we could make the two fields that depend on the `is_discount` to be True to be shown + maybe putting them into a separate group? I guess the first point would make sense, right? Perhaps I'm missing something.